src/extractor/node_based_graph_factory.cpp

```diff
@@ -191,7 +212,11 @@ void NodeBasedGraphFactory::CompressAnnotationData()
     const auto old_size = annotation_data.size();
     // remove all remaining elements
     annotation_data.erase(new_end, annotation_data.end());
-    util::Log() << " graoh compression removed " << (old_size - annotation_data.size())
+
+    // reorder data in the same order
+    std::sort(annotation_data.begin(), annotation_data.end());
```

Comment:
Is this sorting necessary, the std::set is sorted. So we are guaranteed to see IDs in order, when we assign cached_ids[id] = new_id. So for all annotation_ids before and after this algorithm, the IDs should be in order, right?

Reply:
Capturing from chat: since the set is ordered by std::tie instead of the ID, it is. But we should potentially replace this sort by sorting the annotation-ids after unique to ensure that we are future-proof.